annoyed-alligators/socl_media/apps/users/views.py

```diff
@@ -0,0 +1,48 @@
+from django.shortcuts import render, redirect
+from django.contrib.auth import authenticate, login, logout
+from django.contrib.auth.forms import UserCreationForm, AuthenticationForm
+from django.http import HttpResponse
+
+# Create your views here.
+
+
+def signup(request):
+    if request.user.is_authenticated:
+        return redirect('/')
+    if request.method == 'POST':
+        form = UserCreationForm(request.POST)
+        if form.is_valid():
+            form.save()
+            username = form.cleaned_data.get('username')
+            password = form.cleaned_data.get('password1')
+            user = authenticate(username=username, password=password)
+            login(request, user)
+            return redirect('/')
```

Comment:
Instead of logging in the user directly after signup, just redirect to the 'login' page.
So, change this redirect to 'login' and then you can remove the `login(request, user)` line above.